elastic/eui
# [High Contrast Mode] new_final_final(2).psd

## Summary

> [!note]
> This PR merges into a feature branch.

**Me**: I'll just scan through the docs and push up any minor bugfixes directly to the feature branch, I'm sure there's not that many
**Me, 20 commits later**: ...okay yeah I'm just going to go ahead and open a new PR 🥹 

This is the _last_ high contrast PR, I swear(??) - about half component updates and half tweaks/cleanups to docs/themes. [Follow along by commit](https://github.com/elastic/eui/pull/8234/commits) if you dare 🫠 

## QA

- [EuiPanel](https://eui.elastic.co/pr_8234/#/layout/panel#colors-and-corners) with `color="transparent" hasBorder={false}` should no longer render high contrast borders (the same way that `EuiButtonEmpty`s do not render borders)

- [EuiSplitPanel](https://eui.elastic.co/pr_8234/#/layout/panel#split-panels) now renders a dividing border between each inner panel, which attempts to inherit its color from the parent outer panel. I'm not _100%_ sure the amount of code/context required was worth it, but ah well, it looks real nice 😅 
  <img width="377" alt="" src="https://github.com/user-attachments/assets/275524ea-e8c8-4855-8aad-74a6e07d5a14" />

- [EuiEmptyPrompt](https://eui.elastic.co/pr_8234/#/display/empty-prompt)'s border behavior has been updated in light of the transparent panel changes

- [EuiFlyout](https://eui.elastic.co/pr_8234/#/layout/flyout#more-complicated-flyout) had some extra high contrast borders added

- [EuiErrorBoundary](https://eui.elastic.co/pr_8234/#/utilities/error-boundary) was missed during my original component pass and now has high contrast styles
  <img width="400" alt="" src="https://github.com/user-attachments/assets/d41b477a-3949-4eab-8ef2-95effef8549a" /> <img width="400" alt="" src="https://github.com/user-attachments/assets/1f269e9d-09ee-4651-99a5-c07aaae56361" />

- [EuiImage](https://eui.elastic.co/pr_8234/#/display/image#click-an-image-for-a-fullscreen-version)'s hover shadows was missed and is now fixed in high contrast mode

- EuiCodeBlock's border/margin rendering has been fixed [within EuiText](https://eui.elastic.co/pr_8234/#/guidelines/accessibility#naming), and also in Windows high contrast with [scrolling content](https://eui.elastic.co/pr_8234/#/editors-syntax/code#code-block-virtualization) 

- Docs changes: I recommend following along by commit. The general theme was fixing/reducing duplicated borders where it made sense to, but I also made some larger copy updates to the [Colors](https://eui.elastic.co/pr_8234/#/theming/colors/values), [Border](https://eui.elastic.co/pr_8234/#/theming/borders#color), and [Shadows](https://eui.elastic.co/pr_8234/#/theming/more-tokens#shadow) theming pages as they related to high contrast mode.

### General checklist

- Browser QA
- Docs site QA
- Code quality checklist
- Release checklist
    - ~[ ] If applicable, added the **breaking change** issue label (and filled out the breaking change checklist)~ - open to feedback if I yolo'd too hard on this one for the last commit 😅 
- Designer checklist - N/A
